tests/Cache/FileHandlerTest.php

```diff
@@ -145,6 +148,76 @@ public function testWriteWritesCache()
         $this->assertSame($cache->toJson(), $actualCacheJson);
     }
 
+    public function testWriteCacheToDirectory()
+    {
+        $dir = __DIR__.'/../Fixtures/cache-file-handler';
+
+        $handler = new FileHandler($dir);
+
+        $this->setExpectedExceptionRegExp(
+            'Symfony\Component\Filesystem\Exception\IOException',
+            sprintf('#^%s$#', preg_quote('Cannot write cache file "'.realpath($dir).'" as the location exists as directory.', '#')
+        ));
+
+        $handler->write(new Cache(new Signature(
+            PHP_VERSION,
+            '2.0',
+            array(
+                'foo',
+                'bar',
+            )
+        )));
+    }
+
+    public function testWriteCacheToNonWriteableFile()
+    {
+        $file = __DIR__.'/../Fixtures/cache-file-handler/cache-file';
+        if (is_writable($file)) {
+            $this->markTestSkipped(sprintf('File "%s" must be not writeable for this tests.', realpath($file)));
+
+            return;
+        }
+
+        $handler = new FileHandler($file);
+
+        $this->setExpectedExceptionRegExp(
+            'Symfony\Component\Filesystem\Exception\IOException',
+            sprintf('#^%s$#', preg_quote('Cannot write to file "'.realpath($file).'" as it is not writable.', '#')
+        ));
+
+        $handler->write(new Cache(new Signature(
+            PHP_VERSION,
+            '2.0',
+            array(
+                'foo',
+                'bar',
+            )
+        )));
+    }
+
+    public function testWriteCacheFilePermissions()
+    {
+        $file = __DIR__.'/../Fixtures/cache-file-handler/rw_cache.test';
+        @unlink($file);
```

Comment:
why? this file must not exist before test execution

Reply:
to save people the headache when they manually stopped the test before the file was removed

Reply:
please see other comments about why I used this setup.